### ISSUE TITLE
Crash if image is too small

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#1640]
### Description

In reference to [this](https://github.com/neuropoly/spinalcordtoolbox/pull/1564#issuecomment-360342317) comment, `sct_deepseg_gm` crashes if the image is too small in the XY plane. 

Example of error below:
~~~
sct_deepseg_gm -i t2s.nii.gz

--
Spinal Cord Toolbox (master/92f4635e6b35cceeaa558688820c8c778c39d874)
Running /Users/julien/code/sct/scripts/sct_deepseg_gm.py -i t2s.nii.gz
  pixdim: (0.5, 0.5, 7.499999)
  shape: (49, 54, 9)

Calculate new dimensions...
  new shape: (98, 108, 9)
  affine matrix: 
[[ -4.99996871e-01  -8.49516226e-08   2.65367720e-02   9.42460155e+00]
 [ -3.10067044e-05   4.99923617e-01  -1.31088883e-01  -1.32206888e+01]
 [  1.76884665e-03   8.73931497e-03   7.49880648e+00   2.58022213e+00]
 [  0.00000000e+00   0.00000000e+00   0.00000000e+00   1.00000000e+00]]

Calculate affine transformation...
  transfo: 
[[ 0.5   0.    0.   -0.25]
 [ 0.    0.5   0.   -0.25]
 [ 0.    0.    1.    0.  ]
 [ 0.    0.    0.    1.  ]]
Traceback (most recent call last):
  File "/Users/julien/code/sct/scripts/sct_deepseg_gm.py", line 82, in <module>
    run_main()
  File "/Users/julien/code/sct/scripts/sct_deepseg_gm.py", line 72, in run_main
    model_name, int(verbose))
  File "/Users/julien/code/sct/python/lib/python2.7/site-packages/spinalcordtoolbox/deepseg_gm/deepseg_gm.py", line 226, in segment_file
    nii_resampled.shape[1]))
RuntimeError: Image too small (98, 108)
~~~

### Possible solution

pad the image if it is too small.

### Steps to Reproduce

1. go to sct_testing_data/t2s
2. run: `sct_deepseg_gm -i t2s.nii.gz`
3. observe crash

### State of spinalcordtoolbox

commit: 92f4635e6b35cceeaa558688820c8c778c39d874

**Comment by perone** [comment]
This issue is the same as this [issue #34 (axondeepseg)](https://github.com/neuropoly/axondeepseg/issues/34#issuecomment-368744440). The network was trained using a context. A solution for that can be padding, however, the network was trained with this context and this will silently pad the image and input zeros to the network instead of the entire context. In the entire dataset I used (_our internal one and the challenge_), I didn't saw images of this size after resampling to **0.25x0.25mm**, so I would say that when an image has this size, it is usually because it was cropped and is not the original image (was this image that you tested cropped ?).

So there are two paths we can follow from here:

1) Implement the padding;
2) Improve the error message (like done in AxonDeepSeg) to tell the user to use the original image or pad by himself, so the padding will be explicitly done by the user;

In summary, if this image size is from a cropped image and this is uncommon, it would make sense to tell the user to use an uncropped image because the network actually uses the context for prediction. However, if this size is common, I will have to implement a way to pad it after resampling, feed into the network and reconstruct it to the original size again and then resample back.

@jcohenadad What do you think ?

**Comment by jcohenadad** [comment]
ok, i think that going with option 2 is reasonable. I was indeed using a cropped image. It would also be important to mention in the usage/documentation what is the minimum image size required.

**Comment by jcohenadad** [comment]
@perone didn't you fix it in a merged PR?

**Comment by perone** [comment]
Yes this can be closed already.
